packages/stark-ui/src/modules/app-sidebar/components/app-sidebar.component.ts

```diff
@@ -0,0 +1,123 @@
+import { STARK_LOGGING_SERVICE, StarkLoggingService } from "@nationalbankbelgium/stark-core";
+import { StarkAppSidebarOpenEvent } from "../services/app-sidebar-open-event.intf";
+import { StarkAppSidebarService } from "../services/app-sidebar.service";
+
+/**
+ * Name of the component
+ */
+const componentName: string = "stark-app-sidebar";
+
+/**
+ * Component to display the application's sidebar
+ * Only 2 sidebars are allowed: https://github.com/angular/material2/issues/1514
+ */
+@Component({
+	selector: "stark-app-sidebar",
+	templateUrl: "./app-sidebar.component.html",
+	encapsulation: ViewEncapsulation.None
+})
+export class StarkAppSidebarComponent implements OnDestroy, OnInit {
+	/**
+	 * Adds class="stark-app-sidebar" attribute on the host component
+	 */
+	@HostBinding("class")
+	public class: string = componentName;
+
+	/**
+	 * Mode for the left sidebar: either the menu is shown or the regular sidebar
+	 */
+	@Input()
+	public sidenavLeftMode: "regular" | "menu";
+
+	/**
+	 * Reference to the MatSidenavContainer embedded in this component
+	 */
+	@ViewChild("appSidenavContainer")
+	public appSidenavContainer: MatSidenavContainer;
+
+	/**
+	 * Reference to the left MatSidenav embedded in this component
+	 */
+	@ViewChild("appSidenavLeft")
+	public appSidenavLeft: MatSidenav;
+
+	/**
+	 * Reference to the right MatSidenav embedded in this component
+	 */
+	@ViewChild("appSidenavRight")
+	public appSidenavRight: MatSidenav;
+
+	/**
+	 * Subscription to the open sidebar Observable
+	 */
+	public openSidebarSubscription: Subscription;
+
+	/**
+	 * Subscription to the close sidebar Observable
+	 */
+	public closeSidebarSubscription: Subscription;
+
+	/**
+	 * Class constructor
+	 * @param sidebarService - The sidebar service of the application
+	 */
+	public constructor(@Inject(STARK_LOGGING_SERVICE) public logger: StarkLoggingService, public sidebarService: StarkAppSidebarService) {}
+
+	/**
+	 * open sidenav handler
+	 */
+	public onOpenSideNav(event: StarkAppSidebarOpenEvent): void {
+		if (event.mode) {
+			this.sidenavLeftMode = event.mode;
+		}
+		switch (event.sidebar) {
+			case "left":
+				from(this.appSidenavLeft.open()).subscribe(
+					(result: MatDrawerToggleResult) => this.logger.debug(componentName + ": left sidenav " + result),
+					(error: Error) => this.logger.warn(componentName + ": ", error)
+				);
+				break;
+			case "right":
+				from(this.appSidenavRight.open()).subscribe(
+					(result: MatDrawerToggleResult) => this.logger.debug(componentName + ": right sidebar " + result),
+					(error: Error) => this.logger.warn(componentName + ": ", error)
+				);
+				break;
+			default:
+				break;
+		}
+	}
+
+	/**
+	 * close sidenav handler
+	 */
+	public onCloseSideNavs(): void {
+		this.appSidenavContainer.close();
+	}
+
+	/**
+	 * Component lifecycle OnDestroy hook
+	 * Prevent memory leak when component destroyed
+	 */
+	public ngOnDestroy(): void {
```

Comment:
As a convention we put the Angular lifecycle hooks right after the constructor method and in chronological order, this means that the ngOnInit is placed before the ngOnDestriy. This way is easier to find them ;)